cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java

```diff
@@ -73,6 +76,18 @@ public static String getNamespacedBaseDir(NamespacedLocationFactory namespacedLo
     return namespacedLocationFactory.get(namespaceId.toId()).append(logBaseDir).toString();
   }
 
+  public static Location getNamespacedBaseDirLocation(final NamespacedLocationFactory namespacedLocationFactory,
```

Comment:
that's not obvious when looking at this code. Shouldn't access to locations *always* be done under impersonation? It seems like impersonation should be done here, not in FileMetadataManager, right?